File: 137-single-number-ii/single-number-ii.java
class Solution {
    public int singleNumber(int[] nums) {
         Arrays.sort(nums);
         int n = nums.length;
         for(int i=1;i<n;i+=3){
            if(nums[i]!=nums[i-1]) return nums[i-1];
         }
         return nums[n-1];
    }
}

File: 137-single-number-ii/single-number-ii.cpp
class Solution {
public:
    int singleNumber(vector<int>& nums) {
        int n = nums.size();
        unordered_map<int,int>mpp;
        for(int i:nums){
             mpp[i]++;
        }
        for(auto it:mpp){
            if(it.second==1)  return it.first;
        }
        return -1;
    }
};